src/webcore/value.rs

```diff
@@ -51,6 +51,15 @@ impl Reference {
     }
 }
 
+impl PartialEq for Reference {
+    fn eq( &self, other: &Reference ) -> bool {
+        js!( return @{self} === @{other}; ).try_into().unwrap()
+    }
+}
+
+// TODO is this guaranteed to be true? e.g. is it possible to have an f64 that is a Reference ?
```

Comment:
I think it should be safe to have `Eq` implemented for this. (The refids are always going to be integers.)

Reply:
With my implementation of `PartialEq`, it's not comparing refids, it's using JavaScript's `===`, which is why I put that comment there.

Reply:
Well, no primitive types should ever end up as `Reference`s, so in either case it should be safe.